src/connect.rs

```diff
@@ -294,6 +296,15 @@ impl Connector {
         self.verbose.0 = enabled;
     }
 
+    #[cfg(unix)]
+    async fn connect_unix_socket<P: AsRef<Path>>(&self, socket: P) -> Result<Conn, BoxError> {
```

Comment:
Drive-by note: not all Unix sockets are addressed as a `pathname`. There are also `abstract` ones (somehow common) and `unnamed` (quite rare), see section `Address Format` at https://man7.org/linux/man-pages/man7/unix.7.html.

As it is a new API, it would be nice not to constrain it too much to filepaths.

Reply:
I only implemented path-based unix sockets, because the [unix\_socket\_abstract feature is still unstable in the rust standard library](https://github.com/rust-lang/rust/issues/85410). Also, I don't know a use-case, where I could test an HTTP-over-abstract-UDS implementation in a real scenario. Unlike the implementation for path-based unix sockets, that I have used a few months for interaction with snapd and tested with dockerd. I only know abstract unix sockets from libqb's IPC, which doesn't use HTTP. And I have never before heard of unnamed unix sockets, so it would be a bad idea for me to implement something for it.

Considering reqwest is still pre-1.0.0, I think we can implement path-based unix sockets for now, without spoiling abstract unix socket support in the future. And abstract unix sockets are somewhat different anyway, because they are only a linux-specific extension and not generally available on `cfg(unix)`.